Problem3.rb

```diff
@@ -0,0 +1,48 @@
+require 'rspec'
+
+def swap(array, i, j)
+
+  temp = array[i]
+  array[i] = array[j]
+  array[j] = temp
+
+end
+
+def contains_duplicates?(array)
+
+  min = 1
+
+  array.each_index do |i|
+
+    temp = array[i]
+    j = temp - min
+
+    while i != j do
+
+      if temp == array[j]
+        return true
+      end
+
+     swap(array, i, j)
+
+      j = temp - min
+
+    end
+
+  end
+
+  return false
+
+end
+
+describe 'Problem3' do
+
+  it 'should find duplicates' do
+
+    puts contains_duplicates?([1, 2, 3, 4, 2])

```

Comment:
Tests should contain asserts

Reply:
I uploaded tests for problem1 and problem2 using assert